snuba/query/snql/expression_visitor.py

```diff
@@ -0,0 +1,205 @@
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
+
+
+class LowPriOperator(Enum):
+    PLUS = "+"
+    MINUS = "-"
+
+
+class HighPriOperator(Enum):
+    MULTIPLY = "*"
+    DIVIDE = "/"
+
+
+class LowPriTuple(NamedTuple):
+    op: LowPriOperator
+    arithm: Expression
+
+
+class HighPriTuple(NamedTuple):
+    op: HighPriOperator
+    arithm: Expression
+
+
+HighPriArithmetic = Union[Node, HighPriTuple, Sequence[HighPriTuple]]
+LowPriArithmetic = Union[Node, LowPriTuple, Sequence[LowPriTuple]]
+
+
+def get_arithmetic_function(
+    operator: Enum,
+) -> Callable[[Expression, Expression, Optional[str]], FunctionCall]:
+    return {
+        HighPriOperator.MULTIPLY: multiply,
+        HighPriOperator.DIVIDE: divide,
+        LowPriOperator.PLUS: plus,
+        LowPriOperator.MINUS: minus,
+    }[operator]
+
+
+def get_arithmetic_expression(
+    term: Expression, exp: Union[LowPriArithmetic, HighPriArithmetic],
+) -> Expression:
+    if isinstance(exp, Node):
+        return term
+    if isinstance(exp, (LowPriTuple, HighPriTuple)):
+        return get_arithmetic_function(exp.op)(term, exp.arithm, None)
+    elif isinstance(exp, list):
+        for elem in exp:
+            term = get_arithmetic_function(elem.op)(term, elem.arithm, None)
+    return term
+
+
+def exp_visit_function_name(left, node: Node, visited_children: Iterable[Any]) -> str:
+    return str(node.text)
+
+
+def exp_visit_column_name(left, node: Node, visited_children: Iterable[Any]) -> Column:
+    return Column(None, None, node.text)
+
+
+def exp_visit_low_pri_tuple(
+    left, node: Node, visited_children: Tuple[LowPriOperator, Expression]
+) -> LowPriTuple:
+    left, right = visited_children
+
+    return LowPriTuple(op=left, arithm=right)
+
+
+def exp_visit_high_pri_tuple(
+    left, node: Node, visited_children: Tuple[HighPriOperator, Expression]
+) -> HighPriTuple:
+    left, right = visited_children
+
+    return HighPriTuple(op=left, arithm=right)
+
+
+def exp_visit_low_pri_op(
+    left, node: Node, visited_children: Iterable[Any]
+) -> LowPriOperator:
+
+    return LowPriOperator(node.text)
+
+
+def exp_visit_high_pri_op(
+    left, node: Node, visited_children: Iterable[Any]
+) -> HighPriOperator:
+
+    return HighPriOperator(node.text)
+
+
+def exp_visit_arithmetic_term(
+    left, node: Node, visited_children: Tuple[Any, Expression, Any]
+) -> Expression:
+    _, term, _ = visited_children
+
+    return term
+
+
+def exp_visit_low_pri_arithmetic(
+    left, node: Node, visited_children: Tuple[Any, Expression, Any, LowPriArithmetic],
+) -> Expression:
+    _, term, _, exp = visited_children
+
+    return get_arithmetic_expression(term, exp)
+
+
+def exp_visit_high_pri_arithmetic(
+    left, node: Node, visited_children: Tuple[Any, Expression, Any, HighPriArithmetic],
+) -> Expression:
+    _, term, _, exp = visited_children
+
+    return get_arithmetic_expression(term, exp)
```

Comment:
I don't think you need both these methods. they do the same. You can make the visitor call directly `get_arithmetic_expression`.